docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+
+### Storage component
+
+![]()
+
+*<center/> Figure 1.8 Class diagram of Storage </center> <br/></br>*
+
+The Storage component is responsible for saving and retrieving Zoomaster data to and from an external text file.
+
+It uses `Gson` library to encode temporary data from Temp List into a HTML format. Then it writes the encoded data to an external text file. On the other hand, it decodes the HTML format from the external text file and update the Temp List of Zoomaster
+
+The only class carrying out the component's function is the `Storage` class.
+
+Its main roles are:
+
+* Store Zoomaster data to an external text file for long term storage
+* Retrieve Zoomaster data on Initialization
+* Return error messages to the users during extraction or writing
+
+### Local Files component
+
+![]()
+*<center/> Figure 1.9 Class diagram of Local Files </center> <br/></br>*
+
+
+The Local Files component is where Zoomaster's long term storage of data is kept
+
+Its main role is:
+
+* Store Zoomaster data
+
+### Exceptions component
+
+![]()
+
+*<center/> Figure 1.10 Class diagram of Exceptions </center> <br/></br>*
+
+The Exceptions component is responsible for responding to the different errors different components of Zoomaster sends back to the Main function.
+
+It extends the `Exception` class and uses it to catch unique exceptions thrown by different components of Zoomaster.
+
+It consists of  `ZoomasterException` and `ZoomasterExceptionType` classes.
+
+Its main role is:
+
+* Create unique exceptions thrown by different components to signal the Main function what error has occured
+
+
+## **Implementation**
+
+This section explains the implementations of Zoomaster features. It goes through the step-by-step proccess, expected outcomes of each feature and the design considerations.
+
+### Bookmark, Timetable and Planner modes feature (TYS)
+
+Zoomaster has three modes for users to interact in. First, bookmark mode has the list of bookmarks with links to online resources. 
+Secondly, timetable mode has a list of timetable slots. Lastly, planner mode which helps users plan their timetable. 
+To simplify input commands for users, all lists has the same keywords for adding, deleting, and showing items in the lists. 
+Hence by having seperating both list into different modes allows both lists to access the same keywords without causing conflicts when parsing commands.
+
+#### Implementation
+
+This feature extends Command class with a way to toggle between different modes of Zoomaster. The integer variable used to control the modes is stored in the Parser class called "programMode". Additionally, it implements the following operation:
+* getModeFromCommand() - Decodes the command sent by the users to figure out which mode the user wants to move to.
+
+Given below is a sequential diagram of how changing between modes occur.
+
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/ChangeModeCommand%20seq%20dia.png?raw=true) <br/><br/>
```

Comment:
Nice diagram! But perhaps aligning the [setToMode is __] on the left will increase readability?